rotkehlchen/globaldb/handler.py

```diff
@@ -237,6 +237,20 @@ def cleanup(self) -> None:
         if self._packaged_db_conn is not None:
             self._packaged_db_conn.close()
 
+    @staticmethod
+    def _wal_checkpoint(conn: DBConnection) -> None:
+        """
+        Commit wal file. If database is locked we ignore it. This is needed when attacking
+        databases and we want to have up to date information
+        """
+        try:
+            conn.execute('PRAGMA wal_checkpoint(PASSIVE);')
+        except sqlite3.OperationalError as e:
+            if 'database table is locked' in (err := str(e)):
+                log.warning(f'Could not commit globaldb wal file. Ignoring. {err}')
```

Comment:
are you sure this is safe to continue? Won't subsequent DB conn.execute() calls also fail with the same error?

Reply:
No, they don't fail. The reason is that the other operation that is operating in the database finishes and unlocks the database

Reply:
One thing that I couldn't get working is that the initialization of the connection accepts a timeout argument that tells how much python should wait before raising OperationalError if the database is locked. With pragma seems that it doesn't work (?) https://docs.python.org/3/library/sqlite3.html#sqlite3.connect

Reply:
About this I'm not coming up with it. I reproduced as joseph found it and if it fails you can query again the endpoint or do any globaldb query and it works fine